code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -26,126 +26,62 @@
 
 	if(!istype(item, /obj/item/skeleton_key) || spawned_loot)
 		return FALSE
-	var/loot = rand(1,25)
+	var/loot = rand(1,18)
 	switch(loot)
-		if(1 to 2)
-			new /obj/item/disk/design_disk/modkit_disc/resonator_blast(src)  //Doubled chance to receive upgrade disk that is directly relevant to mining
-		if(3 to 4)
-			new /obj/item/disk/design_disk/modkit_disc/rapid_repeater(src)
-		if(5 to 6)
-			new /obj/item/disk/design_disk/modkit_disc/mob_and_turf_aoe(src)
-		if(7 to 8)
-			new /obj/item/disk/design_disk/modkit_disc/bounty(src)
-		if(9)
-			new /obj/item/borg/upgrade/modkit/lifesteal(src)
-		if(10)
-			new /obj/item/shared_storage/red(src)
-		if(11)
+		if(1)
 			new /obj/item/clothing/glasses/godeye(src)
-		if(12)
+		if(2)
 			new /obj/item/reagent_containers/glass/bottle/potion/flight(src)
-		if(13)
+		if(3)
 			new /obj/item/pickaxe/diamond(src) //Ashwalkers exist. This is actually a great drop for them
-		if(14)
+		if(4)
 			new /obj/item/rod_of_asclepius(src)
-		if(15)
+		if(5)
 			new /obj/item/organ/heart/cursed/wizard(src)
-		if(16)
+		if(6)
 			new /obj/item/ship_in_a_bottle(src)
-		if(17)
+		if(7)
 			new /obj/item/jacobs_ladder(src)
-		if(18)
+		if(8)
 			new /obj/item/warp_cube/red(src)
-		if(19)
+		if(9)
 			new /obj/item/wisp_lantern(src)
-		if(20)
+		if(10)
 			new /obj/item/immortality_talisman(src)
-		if(21)
+		if(11)
 			new /obj/item/gun/magic/hook(src)
-		if(22)
+		if(12)
 			new /obj/item/book_of_babel(src)
-		if(23)
+		if(13)
 			new /obj/item/clothing/neck/necklace/memento_mori(src)
-		if(24)
+		if(14)
 			new /obj/item/reagent_containers/glass/waterbottle/relic(src)
-		if(25)
+		if(15)
 			new /obj/item/reagent_containers/glass/bottle/necropolis_seed(src)
+		if(16)
+			new /obj/item/borg/upgrade/modkit/lifesteal(src)
+		if(17)
+			new /obj/item/shared_storage/red(src)
+		if(18)
+			new /obj/item/staff/storm(src)
 	spawned_loot = TRUE
 	qdel(item)
 	to_chat(user, "<span class='notice'>You disable the magic lock, revealing the loot.</span>")
 	return TRUE
 
-/obj/structure/closet/crate/necropolis/tendril/can_open(mob/living/user, force = FALSE)
+/obj/structure/closet/crate/necropolis/can_open(mob/living/user, force = FALSE)
 	if(!spawned_loot)
 		return FALSE
 	return ..()
```

Comment:
Aren't you just making necro chesses that aren't from tendrils impossible to open?

Reply:
Necro chests only come from three locations: Tendrils, Bosses and a sliding puzzle (which now spawns a tendril chest) 
They all require keys to open, but are not impossible to open? 

Is there some additional source of necro chest I'm not thinking of?

Reply:
I have combed the code and confirmed two chests I have locked that I didn't necessarily intend to:
* The sliding puzzle chest is now locked and cannot be opened
* The empty necropolis chest that a Warping Gold slime can generate is locked and cannot be opened. 

These are the only instances of other necro chests, and I'll address those shortly.